app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -132,22 +129,28 @@ class AgendaRepositoryImpl(
         }.map {
             it.id
         }
-        val syncAgendaRequest = SyncAgendaRequest(emptyList(), taskDeleteIds, reminderDeleteIds)
-        when (getAuthResult { agendaApi.syncAgendaItems(syncAgendaRequest) }) {
-            is AuthResult.Authorized -> {
-                reminderDeleteIds.map {
-                    db.getReminderDao().deleteModifiedReminderById(it)
+
+        if (reminderDeleteIds.isNotEmpty() || taskDeleteIds.isNotEmpty()) {
+            val syncAgendaRequest = SyncAgendaRequest(emptyList(), taskDeleteIds, reminderDeleteIds)
+            when (getAuthResult { agendaApi.syncAgendaItems(syncAgendaRequest) }) {
```

Comment:
I'd personally avoid putting such longer functions in a single if or when statement, as it's harder to read then. I'd instead save the result of `getAuthResult` in a variable here and then use that in when. That way you can also apply different formatting